src/lib.rs

```diff
@@ -1205,6 +1245,9 @@ impl Build {
         if !msvc || !is_asm || !is_arm {
             cmd.arg("-c");
         }
+        if self.cuda {
+            cmd.arg("--device-c");
+        }
```

Comment:
Additional thought. And I suppose it's rather a question to @YoshikawaMasashi. I've effectively taken this from #508, but the flag is not **generally** required, only when you compile multiple modules. For this reason it might be more appropriate to shift the burden to application and suggest to add `.flag("--devce-c")` when constructing the compilation. Alternatively one can make the addition contingent on amount of files being compiled.

Reply:
I'll have to defer to your judgement on this, I don't think I know enough about cuda to know what the best option is here.

Reply:
I went for additional `self.files.len() > 1`.